docs/glossary.rst

```diff
@@ -0,0 +1,324 @@
+      file to build wheels in dependency order. Unlike :term:`bootstrap`, it does not
+      perform dependency discovery—it simply builds each package in the specified order.
+      See :doc:`/using` for usage details.
+
+   build tag
+      A numeric prefix added to :term:`wheel` filenames (e.g., ``-0-`` in
+      ``package-1.0.0-0-py3-none-any.whl``) to differentiate wheels built by fromager
+      from upstream wheels. This follows the wheel filename convention from
+      :pep:`427`.
+
+   build-backend dependency
+      A dependency returned by :pep:`517` build backend hooks like
+      ``get_requires_for_build_wheel()``. These are additional tools needed beyond
+      the :term:`build-system dependencies <build-system dependency>` to build a wheel.
+      Examples include ``cython`` for Cython extensions or ``numpy`` for packages
+      compiling against NumPy headers. See :doc:`/concepts/dependencies`.
+
+   build-sdist dependency
+      A dependency needed specifically for building a :term:`source distribution`.
+      Returned by the :pep:`517` ``get_requires_for_build_sdist()`` hook.
+      See :doc:`/concepts/dependencies`.
+
+   build-system dependency
+      A dependency listed in ``pyproject.toml`` under ``[build-system].requires``,
+      as defined by :pep:`518`. These are the foundational build tools (like
+      ``setuptools``, ``flit-core``, or ``maturin``) installed before any build
+      backend hooks are called. See :doc:`/concepts/dependencies`.
+
+   built distribution
+      A package format ready for installation without requiring a build step.
+      :term:`Wheels <wheel>` are the standard built distribution format in Python.
+      Contrast with :term:`source distribution`.
+
+   candidate
+      A potential version of a package discovered during :term:`resolution`. Candidates
+      are evaluated against :term:`version specifiers <specifier>` and
+      :term:`constraints` to select the best matching version.
+
+   canonical name
+      The normalized form of a Python package name, computed using
+      ``packaging.utils.canonicalize_name()``. All letters are lowercase and runs of
+      hyphens, underscores, and periods are replaced with a single hyphen (e.g.,
+      ``My_Package`` becomes ``my-package``). See :ref:`canonical-distribution-names`.
+
+   constraints
+      Version specifications that control package :term:`resolution`. Provided via a
+      ``constraints.txt`` file, they ensure specific versions are used or avoided
+      during builds. Unlike :term:`requirements <requirement>`, constraints only
+      apply when a package is already needed. See :doc:`/files` for format details.
+
+   cyclic dependency
+      A circular dependency where packages depend on each other, forming a loop
+      (e.g., A depends on B, B depends on C, C depends on A). See
+      :term:`build-time cycle` and :term:`install-time cycle`.
+
+   build-time cycle
```

Comment:
These next 2 break alphabetical ordering.

We could fix that by sorting the list, or by removing these as separate terms and just mentioning that cyclic dependencies can happen in build dependencies or in installation dependencies.